src/viam/services/vision/vision.py

```diff
@@ -1,29 +1,131 @@
 import abc
+import sys
 from typing import Any, Final, List, Mapping, Optional, Union
 
 from PIL import Image
 
-from viam.media.video import RawImage
+from viam.media.video import RawImage, ViamImage
 from viam.proto.common import PointCloudObject
-from viam.proto.service.vision import Classification, Detection
+from viam.proto.service.vision import Classification, Detection, GetPropertiesResponse
 from viam.resource.types import RESOURCE_NAMESPACE_RDK, RESOURCE_TYPE_SERVICE, Subtype
 
+if sys.version_info >= (3, 10):
+    from typing import TypeAlias
+else:
+    from typing_extensions import TypeAlias
+
 from ..service_base import ServiceBase
 
 
+class CaptureAllRequest:
+    """
+    CaptureAllRequest represents the collection of fields you can ask the
+    CaptureAllFromCamera method to return. If you request something that the service
+    cannot provide, a warning message will be generated, and the service will not return the
+    requested thing.
+    """
+    def __init__(self, return_image=False, return_classifications=False,
+                 return_detections=False, return_object_point_clouds=False):
+        """
+        Args:
+            return_image (bool): Return the image from the GetImage of the underlying camera.
+            Default is False.
+            return_classifications (bool): Return the classifications from GetClassifications.
+            Default is False.
+            return_detections (bool): Return the detections from GetDetections.
+            Default is False.
+            return_objet_point_clouds (bool): Return the point cloud objects from GetObjectPointClouds.
+            Default is False.
+
+        Returns:
+            None
+        """
+        self.return_image: bool = return_image
+        self.return_classifications: bool = return_classifications
+        self.return_detections: bool = return_detections
+        self.return_object_point_clouds: bool = return_object_point_clouds
+
+
+class CaptureAllResult:
+    """
+    CaptureAllResult represents the collection of things that you have requested from the
+    CaptureAllFromCamera method. This is used most often for visualization purposes, since normally,
+    returning the image on every call to a classifier/detector/etc would be costly and unnecessary.
+    """
+    def __init__(self, image=None, classifications=None, detections=None, objects=None, extra={}):
```

Comment:
I set the default results to `None` rather than the empty list to distinguish between "there was no request for classifier to return a result" vs. "the classifier was requested, but there were no results"